modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -217,84 +172,102 @@ def get_files_info(ip, _rport, shares, info)
 
     logdata = ''
 
-    list = shares.collect { |e| e[0] }
-    list.each do |x|
-      x = x.strip
-      if (x == 'ADMIN$') || (x == 'IPC$')
+    shares.each do |share|
+      share_name = share[:name].strip
+      if SKIPPABLE_SHARES.include? share_name
+        next
+      end
+
+      if (share_name == USERS_SHARE) && !datastore['SpiderProfiles']
         next
       end
 
       if !datastore['ShowFiles']
-        print_status("Spidering #{x}.")
+        print_status("Spidering #{share_name}.")
+      end
+
+      begin
+        tree = simple.client.tree_connect("\\\\#{ip}\\#{share_name}")
+      rescue RubySMB::Error::UnexpectedStatusCode => e
+        print_error "Error when trying to connect to share #{share_name} - #{e.status_code.name}"
+        print_status "Spidering #{share_name} complete."
+        next
       end
+
       subdirs = ['']
-      if (x.strip == 'C$') && datastore['SpiderProfiles']
-        subdirs = profile_options(ip, x)
+      if (DEFAULT_SHARES.include? share_name) && datastore['SpiderProfiles']
+        subdirs = profile_options(tree, share)
       end
       until subdirs.empty?
-        depth = subdirs[0].count('\\')
-        if datastore['SpiderProfiles'] && (x == 'C$')
-          if depth - 2 > datastore['MaxDepth']
+        depth = subdirs.first.count('\\')
+
+        if DEFAULT_SHARES.include? share_name
+          if datastore['SpiderProfiles']
+            if (depth - 2) > datastore['MaxDepth']
+              subdirs.shift
+              next
+            end
+          else
             subdirs.shift
             next
           end
-        elsif depth > datastore['MaxDepth']
+        end
+
+        if depth > datastore['MaxDepth']
           subdirs.shift
           next
         end
-        read, write, type, files = eval_host(ip, x, subdirs[0])
+
+        read, write, _type, files = enum_tree(tree, share, subdirs.first)
+
         if files && (read || write)
-          if files.length < 3
+          if files.empty?
```

Comment:
`enum_tree` skips over `.` and `..`, `next if file_name == '.' || file_name == '..'`
Therefore my understanding is that files will either be empty, or it will have actual directories we can iterate over.

Reply:
That's all good, just wanted to make sure that made sense. It makes sense to skip over those if we are just enumerating the tree for actual files and directories and don't want to return extra info.

Reply:
Marking resolved due to comment above and the fact that this makes sense for what the function is intended to do.